config/main.py

```diff
@@ -490,24 +534,37 @@ def _restart_services():
         'nat',
         'sflow',
     ]
-    generated_services_list = _get_sonic_generated_services()
 
-    if generated_services_list is None:
+    num_asic =  _get_num_asic()
+    generated_services_list, generated_multi_instance_services = _get_sonic_generated_services()
+
+    if ((generated_services_list == []) and
+        (generated_multi_instance_services == [])):
         log_error("Failed to get generated services")
         return
 
     if asic_type == 'mellanox' and 'pmon' in services_to_restart:
         services_to_restart.remove('pmon')
 
     for service in services_to_restart:
-        if service + '.service' not in generated_services_list:
-            continue
-        try:
-            click.echo("Restarting service {} ...".format(service))
-            run_command("systemctl restart {}".format(service))
-        except SystemExit as e:
-            log_error("Restart {} failed with error {}".format(service, e))
-            raise
+        if ((service + '.service' in generated_services_list) or
```

Comment:
duplicated code as above. please use function to reduce the duplicated code.

Reply:
looks like the code is duplicated three times

Reply:
updated code as per comment.